packages/mobile/src/tokens/saga.ts

```diff
@@ -337,7 +353,7 @@ export function* fetchTokenBalancesSaga() {
           .toString()
       }
     }
-    yield put(setTokenBalances(tokens))
+    yield call(updateTokenBalances, tokens)
```

Comment:
I think it would be more robust to really watch the changes to `totalTokenBalanceSelector` after actions are dispatched.

Something like this:
```
let prevValue
while (true) {
  const newValue: ReturnType<typeof totalTokenBalanceSelector> = yield select(totalTokenBalanceSelector)
  if (prevValue !== newValue) {
    // Logic when it changes
  }

  yield take()
}
```

It makes it more resilient to future changes. For instance what if there's another place calling `setTokenBalances` or if the logic in `totalTokenBalanceSelector` changes and something else goes into it?

Right now, `totalTokenBalanceSelector` also uses the local currency exchange rate and if it's updated after this call happens, it won't reflect the right value.

Reply:
thank you for the feedback Jean! this makes a lot of sense, i see we also do this for the user traits (i didn't find this before) - i've updated this PR to use your suggested approach :)